src/audio/smart_amp/smart_amp_maxim_dsm.c

```diff
@@ -279,6 +435,29 @@ int smart_amp_init(struct smart_amp_mod_struct_t *hspk, struct comp_dev *dev)
 	return maxim_dsm_init(hspk, dev);
 }
 
+int smart_amp_get_all_param(struct smart_amp_mod_struct_t *hspk,
+			    struct comp_dev *dev)
+{
+	if (maxim_dsm_get_all_param(hspk, dev) < 0)
+		return -EINVAL;
+	return 0;
+}
+
+int smart_amp_get_num_param(struct smart_amp_mod_struct_t *hspk,
+			    struct comp_dev *dev)
+{
+	enum DSM_API_MESSAGE retcode;
+	int cmdblock[1 + DSM_DEFAULT_NUM_CHANNEL];
+
+	/* Get number of parameters */
+	cmdblock[0] = DSM_SET_CMD_ID(0);
```

Comment:
Btw, does cmdblock[0] always have the same usage ? I'm just wondering here if we can have cmdblock[CMD_ID] and use CMD_ID instead of 0 (and same for 1) ?

Reply:
The format of payload is always same. cmblock[0] is reserved to put parameter ID to read.
CMD_ID=0 is special command to read number of parameters. Will add more macros and comment to explain what it is doing here.